drivers/sdhc/sdhci_common.c

```diff
@@ -474,6 +545,45 @@ static void sdhci_prepare_data(struct sdhci_common *sdhci_ctx, struct sdhc_comma
 	sys_write16(transfer_mode, sdhci_ctx->reg_base + SDHCI_TRANSFER_MODE);
 }
 
+static int sdhci_dma_check(struct sdhci_common *sdhci_ctx, struct sdhc_data *data)
+{
+	uint32_t trans_bytes;
+
+	if (!sdhci_ctx->f_use_dma) {
+		return 0;
+	}
+
+#if defined(CONFIG_64BIT)
+	/* Only 32bit DMA supported */
+	if (((uint64_t)data->data >> 32)) {
+		k_panic();
+		return -EIO;
+	}
+#endif
+
+	trans_bytes = data->block_size * data->blocks;
+
+	if (sdhci_ctx->dma_mode == SDHC_DMA_SDMA) {
+		if (trans_bytes > SDHC_SDMA_XFER_MAX) {
+			LOG_ERR("sdhc:sdma: max xfer size overflow max:%08d", SDHC_SDMA_XFER_MAX);
+			return -EIO;
```

Comment:
Note: in this and the following cases, we can continue to transfer in non-DMA mode.

Reply:
XFER mode is selected during init and dynamic change is not supported now. I have no plans adding such feature now.